src/storage/irc_catalog.cpp

```diff
@@ -115,34 +56,59 @@ optional_ptr<SchemaCatalogEntry> IRCatalog::LookupSchema(CatalogTransaction tran
 	return reinterpret_cast<SchemaCatalogEntry *>(entry.get());
 }
 
+optional_ptr<CatalogEntry> IRCatalog::CreateSchema(CatalogTransaction transaction, CreateSchemaInfo &info) {
+	throw NotImplementedException("IRCatalog::CreateSchema not implemented");
+}
+
+void IRCatalog::DropSchema(ClientContext &context, DropInfo &info) {
+	throw NotImplementedException("IRCatalog::DropSchema not implemented");
+}
+
+PhysicalOperator &IRCatalog::PlanInsert(ClientContext &context, PhysicalPlanGenerator &planner, LogicalInsert &op,
+                                        optional_ptr<PhysicalOperator> plan) {
+	throw NotImplementedException("IRCatalog PlanInsert");
+}
+PhysicalOperator &IRCatalog::PlanCreateTableAs(ClientContext &context, PhysicalPlanGenerator &planner,
+                                               LogicalCreateTable &op, PhysicalOperator &plan) {
+	throw NotImplementedException("IRCatalog PlanCreateTableAs");
+}
+PhysicalOperator &IRCatalog::PlanDelete(ClientContext &context, PhysicalPlanGenerator &planner, LogicalDelete &op,
+                                        PhysicalOperator &plan) {
+	throw NotImplementedException("IRCatalog PlanDelete");
+}
+PhysicalOperator &IRCatalog::PlanUpdate(ClientContext &context, PhysicalPlanGenerator &planner, LogicalUpdate &op,
+                                        PhysicalOperator &plan) {
+	throw NotImplementedException("IRCatalog PlanUpdate");
+}
+unique_ptr<LogicalOperator> IRCatalog::BindCreateIndex(Binder &binder, CreateStatement &stmt, TableCatalogEntry &table,
+                                                       unique_ptr<LogicalOperator> plan) {
+	throw NotImplementedException("IRCatalog BindCreateIndex");
+}
+
 bool IRCatalog::InMemory() {
 	return false;
 }
 
 string IRCatalog::GetDBPath() {
-	return internal_name;
```

Comment:
Are we sure returning warehouse is what we want? If I run 

```
ATTACH 'warehouse' as my_icebergdb (
   ....
);
show all tables;
```

Does `warehouse` show up or `my_icebergdb`? 
I think `my_icebergdb` is better in this case, since the warehouse could be a confusing name in some cases (for example R2)

Reply:
It was returning `internal_name`, which was never set
so it returned an empty string, whatever `GetDBPath` was used for

But yea sure I can change it to the name of the catalog 👍

Reply:
It seems to be used here:
```c++
void AttachedDatabase::Close() {
	D_ASSERT(catalog);
	if (is_closed) {
		return;
	}
	is_closed = true;

	if (!IsSystem() && !catalog->InMemory()) {
		db.GetDatabaseManager().EraseDatabasePath(catalog->GetDBPath());
	}
```

So I think we want it to be empty actually

Reply:
ahh
```c++
	//! A set containing all attached database path
	//! This allows to attach many databases efficiently, and to avoid attaching the
	//! same file path twice
	case_insensitive_set_t db_paths;
```

This is necessary to prevent attaching the catalog twice
I'll look into what it should actually be

Reply:
I think `warehouse` is actually correct:
```c++
attach_options.warehouse = info.path;
```

```c++
	if (options.db_type.empty()) {
		InsertDatabasePath(context, info.path, attached_db->name);
	}
```
```c++
	bool path_exists;
	{
		lock_guard<mutex> path_lock(db_paths_lock);
		path_exists = db_paths.find(path) != db_paths.end();
	}
```